### ISSUE TITLE
code execution bug

### ISSUE THREAD

**Issue by Thermatix** [opal/opal#1286]
I'm not sure if this is a bug or not but when I do:
```ruby
require './example'`
```
 the code in file example.rb doesn't seem to execute 
but if i do 
```ruby
`require_relative './example'`
```
 it does, or have I missed somthing?

**Comment by wied03** [comment]
I can confirm that MRI lets you do `require './example'`. In that case, it seems to function like `require_relative` and does not look for 'example' in the load path, it looks relative to the requiring file.

**Comment by elia** [comment]
Actually `require_relative` has a slightly different behavior, it requires relative from the file in which it's used (vs. the current dir). E.g.:

### `require "./foo"`

```
⤑ cd /tmp/                                                                     ~
⤑ echo "p 1" > foo.rb                                                       /tmp
⤑ mkdir bar                                                                 /tmp
⤑ echo "p 2" > bar/foo.rb                                                   /tmp
⤑ echo "require './foo'" > bar/baz.rb                                       /tmp
⤑ ruby bar/baz.rb                                                           /tmp
1
⤑ echo "require_relative './foo'" > bar/qux.rb                              /tmp
⤑ ruby bar/qux.rb                                                           /tmp
2
⤑       
```

While running opal code current dir info is seldom available, and often doesn't make sense at all (e.g. in a browser), but maybe we could assume "." and support that too, so that `require "./foo"` is the same as `require "foo"`.

**Comment by wied03** [comment]
@elia - I saying that require's behavior is unique in that case. Here's an example:

Let's say I have this code. example.rb looks like this
```ruby
puts 'hello there'
```
When I run `ruby foo.rb` from the junk directory, I get a predictable cannot load such file -- ./example error

When I run `ruby ../foo.rb` from the stuff directory, I get `hello there`. Different behavior than `require_relative`.
![image](https://cloud.githubusercontent.com/assets/297123/14619481/fc66ecb2-0574-11e6-9606-362ba9971a45.png)

**Comment by elia** [comment]
Yeah, sorry, I guess we all agree, I initially misunderstood your reply 😊
